monasca_events_api/tests/first_test.py

```diff
@@ -12,10 +12,246 @@
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+            self.assertFalse(1, msg="Unauthorized Access, should fail but passed")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPUnauthorized)
+            self.assertEqual(e.status, '401 Unauthorized')
+
+    @mock.patch('monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_get_with_eventid_dbdown(self, helper_tenant_id, helpers_validate, mysqlRepo):
+        """GET method when DB Down with event_ID"""
+        mysqlRepo.connect.side_effect = RepositoryException("Database\
+                                                             Connection Error")
+        helpers_validate.validate_authorization.return_value = True
+
+        helper_tenant_id.get_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        event_id = "0ab1ac0a-2867-402d-83c7-d7087262470c"
+
+        eventsObj = EventsSubClass()
+        eventsObj._events_repo = EventsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_get(self._generate_req(), res, event_id)
+            self.assertFalse(1, msg="Database Down, GET should fail but succeeded")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPInternalServerError)
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch('monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_bad_request(self, tenantid, json, http, event, validate, kafka):
+        """POST method with bad request body  """
+        json.return_value = None
+        validate.return_value = True
+        http.side_effect = falcon.HTTPBadRequest('Bad request',
+                                                 'Request body is not valid JSON')
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+            assertFalse(1, msg="Get Method should fail but succeeded, bad request sent")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPBadRequest)
+            self.assertEqual(e.status, '400 Bad Request')
```

Comment:
It might be better if you mocked the helpers library directly here and added side effects to the functions you care about or used a better name than http for the read_http_resource function.